### PR TITLE
OSASINFRA-3644: openstack: allow to pre-define a floating IP for Ingress router-default

### PR DESCRIPTION
**Pull request by EmilienM**
**What this PR does / why we need it**:

Leverage a new feature in the Cluster Ingress Operator where we can now pre-define a floating IP for the Default Ingress Controller.
This will allow customers to pre-create a floating IP, update their DNS and then deploy a HostedCluster. This floating IP will be the load balancer IP for the `router-default` Service created by the Ingress controller.

Feature: [OSASINFRA-3644](https://issues.redhat.com//browse/OSASINFRA-3644)

Note: this contains a workaround commit that will be reverted as soon as we bump openshift/api here.

### CODE REVIEW

**Comment by jparrill** [comment]
Hey @EmilienM let me know when this is ready for review.

**Comment by EmilienM** [comment]
@jparrill thanks a lot. I want to get https://github.com/openshift/cluster-ingress-operator/pull/1147 merged before I update this one and ask for review. However nothing much will change in this PR except the API which changed a bit since the last patchset. So feel free to give early feedback. Thanks

**Comment by EmilienM** [comment]
This worked as expected:

```
HostedCluster:
(...)
    platform:
      openstack:
        ingressFloatingIP: 38.102.83.47
```

And the service/router-default:
```
spec:
  allocateLoadBalancerNodePorts: true
  clusterIP: 172.31.82.11
  clusterIPs:
  - 172.31.82.11
  externalTrafficPolicy: Local
  healthCheckNodePort: 32015
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  loadBalancerIP: 38.102.83.47
  ports:
  - name: http
    nodePort: 31547
    port: 80
    protocol: TCP
    targetPort: http
  - name: https
    nodePort: 31737
    port: 443
    protocol: TCP
    targetPort: https
  selector:
    ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - ip: 38.102.83.47
      ipMode: VIP
```

The job will fail because the CSI is not 100% in place, we need this PR merged:
https://github.com/openshift/hypershift/pull/4936

**Comment by EmilienM** [comment]
e2e-openstack-conformance-ingress-fip can't pass right now because we need https://github.com/openshift/csi-operator/pull/321.

**Comment by EmilienM** [comment]
/test e2e-openstack
e2e-openstack-conformance-ingress-fip can't pass right now because we need https://github.com/openshift/csi-operator/pull/321 but it has proven good results, see https://github.com/openshift/hypershift/pull/4835#issuecomment-2490007570

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go [hunk 1]
```diff
@@ -80,6 +81,24 @@ func ReconcileDefaultIngressController(ingressController *operatorv1.IngressCont
...
 			},
 		}
+	case hyperv1.OpenStackPlatform:
+		ingressController.Spec.EndpointPublishingStrategy = &operatorv1.EndpointPublishingStrategy{
+			Type: operatorv1.LoadBalancerServiceStrategyType,
+			LoadBalancer: &operatorv1.LoadBalancerStrategy{
+				Scope: loadBalancerScope,
+				ProviderParameters: &operatorv1.ProviderLoadBalancerParameters{
+					Type: operatorv1.OpenStackLoadBalancerProvider,
+					// TODO(emilien): add the field once bumped openshift/api and also remove `ReconcileDefaultIngressControllerWithUnstructured`.
```
Comment (bryan-cox) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go:91]: Does this need uncommented?

Comment (EmilienM) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go:91]: I suggest we leave it like this, I made it a separate commit so we can revert it once we bump the openshift/api here.

Comment (EmilienM) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go:91]: I left it here, purely for "documentation" that what we in-fine expect to be the code.

Comment (EmilienM) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go:91]: I'm happy to remove the whole commented block.

**Comment by EmilienM** [comment]
Could we please re do the `/override ci/prow/e2e-aks` ?

**Pull request merged**